sdv/constraints/utils.py

```diff
@@ -126,22 +129,26 @@ def check_nans_row(row):
         return 'None'
 
 
-def add_nans_column(table_data, list_column_names):
-    """Add a categorical column to the table_data indicating where NaNs are.
+def compute_nans_column(table_data, list_column_names):
+    """Compute a categorical column to the table_data indicating where NaNs are.
 
     Args:
         table_data (pandas.DataFrame):
             The table data.
         list_column_names (list):
             The list of column names to check for NaNs.
+
+    Returns:
+        A dict with the column name as key and the column indicating where NaNs are as value.
+        Empty dict if there are no NaNs.
     """
     nan_column_name = '#'.join(list_column_names) + '.nan_component'
```

Comment:
can we pass the name here so we know it is the same instead of recomputing it?

Reply:
Yes I pass the name as an attribute of the class, so we no longer need to recompute it. [8d39f6a](https://github.com/sdv-dev/SDV/pull/1399/commits/8d39f6a89cf3ba2480e3640d8b87131cd658e7a9)